.github/workflows/main_win.yaml

```diff
@@ -186,6 +229,49 @@ jobs:
         with:
           type: onhost
           os: windows
+      - name: Upload On Device Test Artifacts
+        if: |
+          matrix.config == 'devel' &&
+          fromJSON(needs.initialize.outputs.on_device_test).enabled == true &&
+          (
+            github.event_name != 'pull_request' ||
+            contains(github.event.pull_request.labels.*.name, 'on_device')
+          )
+        uses: ./.github/actions/upload_test_artifacts
+        with:
+          type: ondevice
+          os: linux
+
+  # Runs on-device integration and unit tests.
+  on-device-test:
+    needs: [initialize, build]
+    # Run ODT when on_device label is applied on PR.
+    # Also, run ODT on push and schedule if not explicitly disabled via repo vars.
+    if: always()
+    runs-on: [self-hosted, odt-runner]
+    name: ${{ matrix.name }}_on_device_${{ matrix.shard }}
+    container: ${{ needs.docker-unittest-image.outputs.docker_unittest_tag }}
```

Comment:
We can and should run on-device-test job directly on the runner.  This is going to be addressed in https://github.com/youtube/cobalt/pull/2432, so we can remove this line altogether.